### PR TITLE
chore/NS-79_update-cluster-as

### PR DESCRIPTION
**Pull request by PeterNMcArthur**

### CODE REVIEW

**Review by yankcrime — COMMENTED** [review]
What needs to be added / changed here is a new version defining with the corresponding Helm chart version, instead of updating this image.tag value inline.  The KubernetesClusterApplicationBundle then also needs updating.

File: charts/unikorn/templates/applications.yaml [hunk 1]
```diff
@@ -287,7 +287,7 @@ spec:
     - name: extraArgs.enforce-node-group-min-size
       value: 'true'
     - name: image.tag
-      value: v1.27.2
+      value: v1.30.0
```
Comment (spjmurray) [inline charts/unikorn/templates/applications.yaml:290]: Why this is here in the first place, I cannot tell you, but if you look up ^^^ what you should be striving to alter is the _chart version_ e.g.

```
[Wed 15 May 09:10:01 BST 2024] simon@symphony ~/src/github.com/unikorn-cloud/client-go helm search repo cluster-autoscaler
NAME                                              	CHART VERSION	APP VERSION	DESCRIPTION                                       
cluster-autoscaler/cluster-autoscaler             	9.37.0       	1.30.0     	Scales Kubernetes worker nodes within autoscali...
```

This will usually upgrade the image version implicitly, so not only do you get the new binary, you get the supporting infrastructure for that binary the chart provides.

So if you follow all the links you get to https://github.com/kubernetes/autoscaler/blob/cluster-autoscaler-chart-9.37.0/charts/cluster-autoscaler/values.yaml#L239 so delete this tag override entirely and upgrade the chart!

Basically what @yankcrime has alluded to with a bit more detail 😸

**Review by spjmurray — CHANGES_REQUESTED** [review]

**Review by spjmurray — COMMENTED** [review]
I really cannot see what the fuck has actually changed here 😉 

While I do agree we should have a YAML linter, it should apply to everything and not just selectively, and be guarded by a CI job.  Obviously dropping in `yamlfmt` isn't going to fly as this is mustache code, and support for that is a bit ropey!  I may be able to do some remove, add sentinels, format, add back black magic...
